detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/Analyzer.kt

```diff
@@ -145,6 +154,11 @@ internal class Analyzer(
     }
 }
 
+internal fun extractRuleName(key: String): Rule.Name? =
+    runCatching { Rule.Name(key.split("/", limit = 2).first()) }.getOrNull()
```

Comment:
Is there a way to get better DX here? In what cases can this throw and return null? If I read it correctly it'll simply ignore the rule. Should we fail when config is incorrect rather than let the user scratch their head why things are "not working". I guess the Rule.Name validation was there before, but now the runCatching can fail due to 2 reasons. Consider to get back "original" behavior:

```suggestion
    val name = key.split("/", limit = 2).first()
    return runCatching { Rule.Name(name) }.getOrNull()
```

Reply:
Are we not allowing `/` in original rule id? Also instead of `runCatching { Rule.Name(key.split("/", limit = 2).first()) }.getOrNull()` can we do `Rule.Name(key.split("/", limit = 2).firstOrNull()`

Reply:
Same, I did these changes at #7468